### DISCUSSION TITLE
Segmentation Fault

### DISCUSSION THREAD

**Discussion by mpalaciosc** [IIC2133-PUC/2024-1#119] [Tarea 1] [answered]
Holaa! Me esta pasando que cuando corro el ejecutable de la parte 2 sin valgrind (./hsearch test/test.txt i.txt) me sale Segmentation Fault, pero cuando lo corro con valgrind (valgrind ./hsearch test/test.txt i.txt), el codigo lo corre bien, escribiendo correctamente en el output.

Alguna razon de porque pueda pasar esto?
Influira en la correcion de la tarea, o eso tambien lo corren con valgrind?

Si es que sirve de algo, mi output cuando corro con valgrind es:

==750== Memcheck, a memory error detector
==750== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==750== Using Valgrind-3.15.0 and LibVEX; rerun with -h for copyright info
==750== Command: ./hsearch test/test.txt i.txt
==750==
==750==
==750== HEAP SUMMARY:
==750==     in use at exit: 0 bytes in 0 blocks
==750==   total heap usage: 34,671 allocs, 34,671 frees, 1,622,200 bytes allocated
==750==
==750== All heap blocks were freed -- no leaks are possible
==750==
==750== For lists of detected and suppressed errors, rerun with: -s
==750== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)

**Comment by benjavicente** [comment, marked as answer]
Está raro. Prueba arreglando los warnings que salen al compilar antes de ver bien que está pasando. Valgrind corre en un entorno levemente distinto, que probablemente cause que no muestre el error dado lo que te ha advertido el compilador.

**Reply by mpalaciosc** [reply]
Buenisimo gracias! Pense que no habia que darle mucha vuelta a los warnings si es que compilaba, pero resolviendolos se me arreglo
